src/Castle.Facilities.AspNetCore/WindsorContainerExtensions.cs

```
@@ -0,0 +1,64 @@
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Castle.Facilities.AspNetCore
+{
+	public static class WindsorContainerExtensions
+	{
+		/// <summary>
+		/// For making types available to the IServiceCollection using 'late bound' factories resolved via Windsor. This makes things like the @Inject directive in Razor work.
+		/// Service must be registered otherwise an ArgumentException will be thrown
+		/// </summary>
+		/// <typeparam name="TService">Service type</typeparam>
+		/// <param name="container">Windsor container which 'activators'/'cross wired' components resolve from</param>
+		/// <param name="serviceCollection">ASP.NET Core service collection from Microsoft.Extensions.DependencyInjection</param>
+		/// <param name="lifestyleType">The lifestyle type that needs to be copied across, only Singleton, Scoped and Transient are supported</param>
+		public static void CrossWired<TService>(this IWindsorContainer container, IServiceCollection serviceCollection,
+			LifestyleType lifestyleType = LifestyleType.Scoped)
+		{
+			container.CrossWired(typeof(TService), serviceCollection, lifestyleType);
+		}
+
+		/// <summary>
+		/// For making types available to the IServiceCollection using 'late bound' factories resolved via Windsor. This makes things like the @Inject directive in Razor work.
+		/// Service must be registered otherwise an ArgumentException will be thrown
+		/// </summary>
+		/// <param name="container">Windsor container which 'activators'/'cross wired' components resolve from</param>
+		/// <param name="serviceType">The type of the service</param>
+		/// <param name="serviceCollection">ASP.NET Core service collection from Microsoft.Extensions.DependencyInjection</param>
+		/// <param name="lifestyleType">The lifestyle type that needs to be copied across, only Singleton, Scoped and Transient are supported</param>
+		public static void CrossWired(this IWindsorContainer container, Type serviceType, IServiceCollection serviceCollection,
```

Why are we exposing this on the container? It goes against how the fluent registration API works please see docs here: https://github.com/castleproject/Windsor/blob/master/docs/fluent-registration-api.md

Typically registration API's are surface via ComponentRegistration<T>.

---

At the very least we should rename this to `RegisterCrossWired`.

---

Am i exposing the container? i am extending the container. So the owner will need to be in control of the container to do this. as mentioned below, this is to allow post registration cross wiring.

---

Recreating the ServiceProvider to support this comes with a severe performance penalty when it comes to memory management.

---

I would rather throw a helpful exception which asks folks to register their `CrossWires` in the right place.

---

Your choice I am happy to do both as long as our docs are in order.